src/components/data_search/DatasetSearchTable.jsx

```diff
@@ -49,8 +51,21 @@ export const DatasetSearchTable = (props) => {
   const [selectedTable, setSelectedTable] = useState(datasetSearchTableTabs.study);
   const [searchTerm, setSearchTerm] = useState('');
 
-  const isFiltered = (filter, category) => (filters[category]).indexOf(filter) > -1;
-  const numSelectedFilters = (filters) => Object.values(filters).reduce((sum, array) => sum + array.length, 0);
+  const isFilteredArray = (filter, category) => (filters[category]).indexOf(filter) > -1;
+
+  const numSelectedFilters = (filters) => {
```

Comment:
This could still use `reduce`, is there a reason not to use it here? It would look something like
```
filters.reduce((sum, category) => sum + isArray(filters[category]) ? filters[category].length : 1, 0)
```

I find `reduce()` clearer to read than using a `for`, as I don't have to read the code to understand how `sum` is generated at the end. It also avoids mutable state which I find easier to think about.

Reply:
A little nuance to your first comment, if the filter is not an array, it does not mean add one, it means check if it is not null, and if it is not null add one